docs/workflow.md

```diff
@@ -47,3 +47,24 @@ It is up to the remediation provider to delete the remediation CR in case the
 node is deleted, and a new one (with a different name) is reprovisioned.
 In that specific scenario the NHC controller can not detect a successful node
 remediation and delete the remediation CR itself.
+
+### Special cases
+
+#### Ignoring unhealthy node condition duration in case node condition changes during remediation
+
+Sometimes the node status changes from one condition to another one during remediation, and both conditions are configured as unhealthy.
+E.g. when a node with condition type `Ready` and status `Unknown` is rebooted, it typically moves to `Ready` = `False` during reboot,
+before finally moving to `Ready` = `True` when done.
+
+This can result in remediation loops: in theory the node has to be considered as healthy while `Ready` = `False`, as long as the configured duration
+didn't expire yet. Subsequently, when the duration expires before the node moves to `Ready` = `True`, the node is considered unhealthy and remediated again.
+
+In order to prevent such remediation loops, NHC will ignore unhealthy nodes which don't have any unhealthy condition with expired duration anymore,
```

Comment:
This sounds a bit tricy to read for me :)
What does "ignore" mean here? NHC will not consider this nodes at all (like nor healthy, nor unhealthy)?

Reply:
Correct, that's exactly the meaning. 
I'm open to suggestions to improve wording 🙂

Reply:
Maybe just this detail, e.g.

> will ignore completely (not healthy, nor unhealthy) nodes ...

but I'm not sure if it's worth it, it is just a nitpik.

/lgtm
holding only because there are some other open threads. Feel free to unhold.
/hold

Reply:
How about:
instead of "NHC will ignore unhealthy nodes which don't have any unhealthy condition with expired duration anymore"
use "Once a node is unhealthy NHC will keep considering that node as unhealthy until all the node's conditions are healthy, regardless of configured unhealthy conditions duration"

Reply:
> NHC will keep considering that node as unhealthy

that's a difference to ignoring nodes and would be wrong IMHO, because we don't continue with escalating remediation 🤷🏼‍♂️